source/particles/particle_handler.cc

```diff
@@ -1561,6 +1573,27 @@ namespace Particles
                 cell_to_neighbor_subdomain.insert(
                   vertex_to_neighbor_subdomain[cell->vertex_index(v)].begin(),
                   vertex_to_neighbor_subdomain[cell->vertex_index(v)].end());
+
+                // If the vertex has one or more coinciding vertices, then
+                // all of these vertices, including the current vertex being
+                // looped over are contained inside the
+                // vertex_to_coinciding_vertex_group Consequently, we loop over
+                // the full content of the coinciding vertex group which will
+                // include the current vertex. When we reach the current vertex,
+                // we skip it to avoid repetition.
+                if (vertex_to_coinciding_vertex_group.find(cell->vertex_index(
+                      v)) != vertex_to_coinciding_vertex_group.end())
+                  {
+                    for (const unsigned int cv : coinciding_vertex_groups.at(
+                           vertex_to_coinciding_vertex_group.at(
+                             cell->vertex_index(v))))
+                      {
+                        if (cv != cell->vertex_index(v))
+                          cell_to_neighbor_subdomain.insert(
+                            vertex_to_neighbor_subdomain[cv].begin(),
+                            vertex_to_neighbor_subdomain[cv].end());
+                      }
+                  }
```

Comment:
This works, but I wonder if the syntax would not be easier if you had used `GridTools::compute_vertices_with_ghost_neighbors` above and store the result of that function in the triangulation_cache. Then you could do the following here:

```suggestion
const auto &vertex_to_ghost_neighbors = GridTools::compute_vertices_with_ghost_neighbors(tria);
...
...
               if (vertex_to_ghost_neighbors.find(cell->vertex_index(
                      v)) != vertex_to_ghost_neighbors.end())
                  {
                       for (const auto &vertex_and_ghost_neighbors : vertex_to_ghost_neighbors.at(cell->vertex_index(v)))
                      {
                         cell_to_neighbor_subdomain.insert(
                            vertex_and_ghost_neighbors.second.begin(),
                            vertex_and_ghost_neighbors.second.end());
                      }
                  }
```

No need to check for duplicate vertex index then.

Reply:
@gassmoeller  I am not sure I find this approach cleaner since it seems a bit more asbtract to me, maybe it is because I don't find the documentation of the compute_vertices_with_ghost_neighbors function very clear. One thing that is unclear to me is does vertex_to_ghost_neighbors contain only the subdomains connected to the duplicated vertices or does it also contain the subdomains connected to the original vertex? the documentation is not very clear about this. In the first case, then this  would also remove the insertion above for the present vertex, am I right?